scripts/launch-s0-instance.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+me=$(basename $0)
+dir=$(readlink -f $(dirname $0))
+elastio_aws_id=537513441174
+security_group="elastio-s0-server"
+default_instance_type="m5d.2xlarge"
+default_instance_name="$(whoami)-s0-server"
+
+usage()
+{
+    echo "Usage examples:"
+    echo "   $me --key-name my_ec2_key --name my_s0_server --type m5d.2xlarge"
+    echo "   $me --k /path/to/key.pem -n my_s0_server -t m5d.2xlarge"
+    echo
+    echo "  -k | --key-name : A name of an aws EC2 key in the PEM format for the SSH connection to the instance."
+    echo
+    echo "  -n | --name     : Optional. Name of an instance. It's shown in the AWS console."
+    echo "                    The default value is \"[user_name]-s0-server\"."
+    echo
+    echo "  -t | --type     : Optional. An instance type. See https://aws.amazon.com/ec2/instance-types/ for more details."
+    echo "                    The default value is \"m5d.2xlarge\"."
+    echo
+    echo "  -h | --help     : Show this usage help."
+}
+
+[ "$1" == "" ] && echo "Script need some arguments!" && usage && exit -1
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -k | --key-name)   shift && key_name=$1 ;;
+        -n | --name)       shift && instance_name=$1 ;;
+        -t | --type)       shift && instance_type=$1 ;;
+        -h | --help)       usage && exit ;;
+        *)                 echo "Wrong arguments!"
+                           usage && exit 15 ;;
+    esac
+        shift
+done
+
+set -e
+
+if [ -z "$key_name" ]; then
+    echo "Please specify the '--key-name' parameter. Otherwise, you won't be able to connect to the running instance via ssh!"
+    echo
+    usage
+    exit 1
+fi
+
+[ -z "$instance_name" ] && instance_name=$default_instance_name
+[ -z "$instance_type" ] && instance_type=$default_instance_type
+
+latest_ami=$(aws ec2 describe-images --owners $elastio_aws_id \
+    --filters "Name=name,Values=s0-ami*" "Name=state,Values=available" \
+    --query "reverse(sort_by(Images, &CreationDate))[:1].ImageId" --output text)
+
+if aws ec2 describe-security-groups | jq '.SecurityGroups[].GroupName' | tr -d '"' | grep -q "$security_group" ; then
```

Comment:
That's a great convenience feature but it should be possible to override this and specify a security group of the caller's choosing.

Reply:
Do you mean add an optional script parameter for a security group name and create a group with this name if it doesn't exist? Or just use a customer's existing group? I'm sure no one has a group with the 61234 port opened.

Reply:
done